### PR TITLE
Correct and extend documentation for lib/y2lan_restart_common.pm

### PR DESCRIPTION
**Pull request by Zaoliang**
see https://progress.opensuse.org/issues/57566

### CODE REVIEW

File: lib/y2lan_restart_common.pm [hunk 1]
```diff
@@ -38,6 +43,13 @@ our @EXPORT = qw(
 );
 my $module_name;
 
+=head2 initialize_y2lan
+
+ initialize_y2lan();
+
+Initialize yast2 lan. Stop firewalld, enable debug for detailed messages and easier detection of restart, clear journal.log
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: You could shorten the line rephrasing these items:
`s/Stop firewalld/Ensure firewalld is stopped/`
`s/enable debug for detailed messages and easier detection of restart/Enable DEBUG/`
`s/clear journal.log/Clear the journal/`

File: lib/y2lan_restart_common.pm [hunk 2]
```diff
@@ -55,14 +67,28 @@ sub initialize_y2lan
     assert_script_run '> journal.log';    # clear journal.log
 }
 
+=head2 open_network_settings
+
+ open_network_settings();
+
+Open yast2 lan overview tab and select first device
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: `s/Open yast2 lan overview tab/Open yast2 lan module, expecting Overview tab/`
Missing what accept_warning_manager_default does

File: lib/y2lan_restart_common.pm [hunk 3]
```diff
@@ -55,14 +67,28 @@ sub initialize_y2lan
...
+    assert_screen 'yast2_lan', 180;    # yast2 lan overview tab
+    send_key 'home';                   # select first device
     wait_still_screen 1, 1;
 }
 
+=head2 close_network_settings
+
+ close_network_settings();
+
+Handle warning popup for firewall and assign it to zone or install yast2 lan packages. Close network settings.
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: `s/firewall/firewalld/`
`s/Close network settings./Close network settings checking for return code 0 on serial line./`

File: lib/y2lan_restart_common.pm [hunk 4]
```diff
@@ -105,16 +138,34 @@ sub check_network_status {
     type_string "\n\n";                                               # make space for better readability of the console
 }
 
+=head2 verify_network_configuration
+
+ verify_network_configuration([$fn], [$dev_name], [$expected_status], [$workaround], [$no_network_check]);
+
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: I think in this case is important to describe the parameters, in particular `$fn`, which is basically a reference to a function with the action to be performed.

File: lib/y2lan_restart_common.pm [hunk 5]
```diff
@@ -197,6 +266,13 @@ sub check_etc_hosts_update {
     set_network;
 }
 
+=head2 handle_Networkmanager_controlled
+
+ handle_Networkmanager_controlled();
+
+Handle Networkmanager controls the network configurations
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: I think here it requires some adjustment, I cannot get it from the description, it seems to be a popup.

File: lib/y2lan_restart_common.pm [hunk 6]
```diff
@@ -209,12 +285,28 @@ sub handle_Networkmanager_controlled {
...
     if (match_has_tag('dhcp-popup')) {
         wait_screen_change { send_key 'alt-o' };
     }
 }
 
+=head2 open_yast2_lan
+
+ open_yast2_lan();
+
+Open yast2 lan, run handle_dhcp_popup() and install and check firewall
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: `s/firewall/firewalld/

File: lib/y2lan_restart_common.pm [hunk 7]
```diff
@@ -1,3 +1,8 @@
+=head1 y2lan_restart_common
+
+YaST logic on Network Restart while no config changes were made
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: What about, `Library for non-destructive testing using yast2 lan`

Comment (Zaoliang) [reply lib/y2lan_restart_common.pm]: Now update like this:

Library for non-destructive testing using yast2 lan. YaST logic on Network Restart while no config changes were made.

File: lib/y2lan_restart_common.pm [hunk 8]
```diff
@@ -1,3 +1,8 @@
+=head1 y2lan_restart_common
+
+Library for non-destructive testing using yast2 lan. YaST logic on Network Restart while no config changes were made.
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: you can delete second part, it is the same.

File: lib/y2lan_restart_common.pm [hunk 9]
```diff
@@ -38,6 +43,21 @@ our @EXPORT = qw(
 );
 my $module_name;
 
+=head2 initialize_y2lan
+
+ initialize_y2lan();
+
+Initialize yast2 lan.
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: better in one line all the comments or if you find some more fancy way in Perl, up to you.

File: lib/y2lan_restart_common.pm [hunk 10]
```diff
@@ -55,14 +75,30 @@ sub initialize_y2lan
     assert_script_run '> journal.log';    # clear journal.log
 }
 
+=head2 open_network_settings
+
+ open_network_settings();
+
+Open yast2 lan module, expecting Overview tab and select first device.
+
+Accept warning for Networkmanager controls network device.
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: I would include this sentence in the previous one.

File: lib/y2lan_restart_common.pm [hunk 11]
```diff
@@ -105,16 +148,44 @@ sub check_network_status {
     type_string "\n\n";                                               # make space for better readability of the console
 }
 
+=head2 verify_network_configuration
+
+ verify_network_configuration([$fn], [$dev_name], [$expected_status], [$workaround], [$no_network_check]);
+
+C<$fn> is basically a reference to the function with the action to be performed.
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: without `basically`

File: lib/y2lan_restart_common.pm [hunk 12]
```diff
@@ -197,6 +286,13 @@ sub check_etc_hosts_update {
     set_network;
 }
 
+=head2 handle_Networkmanager_controlled
+
+ handle_Networkmanager_controlled();
+
+Handle Networkmanager controls the network configurations, confirm if a warning popup for Networkmanager controls networking.
+
```
Comment (jknphy) [inline lib/y2lan_restart_common.pm]: sepparate this line in two

File: lib/y2lan_restart_common.pm (hunk 9 above)
Comment (Zaoliang) [reply lib/y2lan_restart_common.pm]: script/generateLibDocs.sh has problem to handle this shorten sentences without empty line between lines. Anyway this should be clear enough to read.

File: lib/y2lan_restart_common.pm (hunk 10 above)
Comment (Zaoliang) [reply lib/y2lan_restart_common.pm]: Accept warning for Networkmanager controls network device -> in which previous one?

Comment (jknphy) [reply lib/y2lan_restart_common.pm]: it is fine now I think.

**Pull request merged**